esp-hal/src/spi/master.rs

```diff
@@ -1071,9 +1110,10 @@ where
     ///
     /// # Errors
     ///
-    /// The corresponding error variant from [`Error`] will be returned if
+    /// [`Error::FifoSizeExeeded`] or [`Error::Unsupported`] will be returned if
     /// passed buffer is bigger than FIFO size or if buffer is empty (currently
```

Comment:
not to be resolved in this PR but I think in some cases passing zero-length buffers are no-ops and in some cases we return errors - we should be consistent here (IMHO it should be a no-op but something to discuss)